xarray/core/variable.py

```diff
@@ -298,11 +246,12 @@ def convert_non_numpy_type(data):
         data = utils.to_0d_object_array(data)
 
     if isinstance(data, pd.Timestamp):
-        # TODO: convert, handle datetime objects, too
-        data = np.datetime64(data.value, "ns")
+        data = data.to_numpy()
 
+    if isinstance(data, datetime):
+        data = np.datetime64(data)
     if isinstance(data, timedelta):
-        data = np.timedelta64(getattr(data, "value", data), "ns")
+        data = np.timedelta64(data)
```

Comment:
I've tried to drop pd.Series here, but failed. So I'm open to any solution.

Reply:
> I think pandas still converts scalar datetime.datetime and datetime.timedelta objects to nanosecond precision values instead of microsecond precision values

This behavior seems quite bad from pandas. Hopefully there's willingness to fix it upstream?

Reply:
Yeah, I agree, we should inquire about it if there is not an open issue already.